internal/pkg/metrics/kyma.go

```diff
@@ -141,3 +146,68 @@ func calcStateValue(state, newState shared.State) float64 {
 func (k *KymaMetrics) RecordRequeueReason(kymaRequeueReason KymaRequeueReason, requeueType RequeueType) {
 	k.requeueReasonCounter.WithLabelValues(string(kymaRequeueReason), string(requeueType)).Inc()
 }
+
+func (k *KymaMetrics) CleanupNonExistingKymaCrsMetrics(ctx context.Context, kcpClient client.Client) error {
+	currentLifecycleManagerLogs, err := FetchLifecycleManagerMetrics()
+	if err != nil {
+		return fmt.Errorf("failed to fetch current kyma metrics, %w", err)
+	}
+
+	if len(currentLifecycleManagerLogs) == 0 {
+		return nil
+	}
+
+	kymaCrsList := &v1beta2.KymaList{}
+	err = kcpClient.List(ctx, kymaCrsList)
+	if err != nil {
+		return fmt.Errorf("failed to fetch Kyma CRs, %w", err)
+	}
+	kymaNames := getKymaNames(kymaCrsList)
+	for _, m := range currentLifecycleManagerLogs {
+		currentKymaName := getKymaNameFromLabels(m)
+		if !slices.Contains(kymaNames, currentKymaName) {
+			k.KymaStateGauge.DeletePartialMatch(prometheus.Labels{
+				KymaNameLabel: currentKymaName,
+			})
+		}
+	}
```

Comment:
I think this works, but maybe a set based comparison would read a bit easier and even be a bit more performant? Something like

```go
kymaNames := getKymaNames(kymaCrsList) // returning a "kymaName->true" mapping (map[string]bool)
currentKymaName := getKymaNameFromLabels(m)

if _, exists := kymaNames[currentKymaName]; !exists {
  k.KymaStateGauge.DeletePartialMatch(prometheus.Labels{
    KymaNameLabel: currentKymaName,
  })
}
```